src/os.cc

```diff
@@ -42,8 +42,11 @@
 #      define S_IROTH 0
 #    endif
 
+
 #    ifdef __MINGW32__
 #      define _SH_DENYNO 0x40
+#    else
+       using mode_t = int;
```

Comment:
Let's put `mode_t` in the `fmt::detail` namespace and in the `file` constructor add using `namespace fmt::detail`. Then it will pick up our `fmt::detail::mode_t` if available and the global one otherwise. Also no need to check for `__MINGW32__`.